### PR TITLE
Content build validation

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

Related to https://github.com/department-of-veterans-affairs/va.gov-team/issues/7969

This adds a script which will compare the content builds from `vets-website` and the newly formed [`content-build`](https://github.com/department-of-veterans-affairs/content-build) repo.

## Testing done

In the `content-build` repo, run:

- `yarn build:content --pull-drupal`

In `vets-website`, run the same command.

Once both builds have finished, run:

- `yarn build:compare`

## Screenshots

An example of differing builds (I made this happen by comparing an older build from the standalone content build repo with a content build in `vets-website` that used `--pull-drupal`

image

This is what the output will be when both builds match (they both used identical drupal content):

image

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: script/content-build-compare.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    path.join(__dirname, `../../content-build/build/${buildtype}`),
+    'standaloneContentBuildHash.txt',
+  );
+
+  /* eslint-disable no-console */
+  if (isEqual(websiteContentBuild, standaloneContentBuild)) {
+    console.log('The content builds match!');
+  } else {
+    console.log('The content builds do not match');
+  }
```
Comment (bkjohnson) [inline script/content-build-compare.js:77]: This output feels a bit weak to me.  Open to any suggestions on how to improve it.

File: package.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
     "analyze": "webpack-bundle-analyzer build/vagovprod/generated/stats.json",
     "build": "./script/build.sh",
     "build-analyze": "NODE_ENV=production npm run build -- --buildtype vagovprod --analyzer; npm run analyze",
+    "build:compare": "node script/content-build-compare.js && diff --color websiteContentBuildHash.txt standaloneContentBuildHash.txt",
```
Comment (bkjohnson) [inline package.json:10]: I'm not thrilled by how the node script writes the files and we have a separate bash command to do the diffing, but I wasn't able to get the output to show if I ran the diff inside of the node script.

File: script/content-build-compare.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  const websiteContentBuild = hashBuildOutput(
+    path.join(__dirname, `../build/${buildtype}`),
+    'websiteContentBuildHash.txt',
+  );
+  const standaloneContentBuild = hashBuildOutput(
+    path.join(__dirname, `../../content-build/build/${buildtype}`),
+    'standaloneContentBuildHash.txt',
+  );
+
+  /* eslint-disable no-console */
```
[the remark this answers is not in the document]
Comment (bkjohnson) [reply script/content-build-compare.js:81]: I want to show some kind of output for this.

File: script/content-build-compare.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+const path = require('path');
+const { isEqual } = require('lodash');
+
+// Modeled after https://coderrocketfuel.com/article/recursively-list-all-the-files-in-a-directory-using-node-js
+function getAllFiles(dirPath, arrayOfFiles = []) {
+  const files = fs.readdirSync(dirPath);
+
+  files.forEach(file => {
+    if (fs.statSync(`${dirPath}/${file}`).isDirectory()) {
+      /* eslint-disable no-param-reassign */
```
[the remark this answers is not in the document]
Comment (bkjohnson) [reply script/content-build-compare.js:12]: This was code that I copied from a blog - I didn't want to spend too much time refactoring.

File: package.json (hunk 2 above)
Comment (bkjohnson) [reply package.json:10]: Still unsuccessful :disappointed:

Comment (rianfowler) [reply package.json:10]: This is totally fine- we're not going to be keeping it around once everything is separated and validated.

**Review by rianfowler — APPROVED** [review]
This looks great to me. Did you do any manual testing on it? like deliberately try to break each build to make sure that there's a failure when that happens? i would try editing and removing a file (each in isolation).

**Comment by bkjohnson** [comment]
Yes - in addition to intentionally testing a newer build against an older build, I also modified one of the build files to get a different hash.  This is what that looks like:

image


This is what it looks like if one of the builds is completely missing a file:

image

**Pull request merged**